Mathlib/Combinatorics/SimpleGraph/Hasse.lean

```diff
@@ -122,4 +123,230 @@ theorem pathGraph_two_eq_top : pathGraph 2 = ⊤ := by
   ext u v
   fin_cases u <;> fin_cases v <;> simp [pathGraph, ← Fin.coe_covBy_iff, Nat.covBy_iff_succ_eq]
 
+/-- Hommorphism from smaller path graph to bigger path graph· -/
+@[simps]
+protected def Hom.pathGraph {n m : ℕ} (hnm : n ≤ m) : pathGraph n →g pathGraph m where
+  toFun v := ⟨v.val, trans v.is_lt hnm⟩
+  map_rel' := by simp [pathGraph_adj]
+
+protected theorem Hom.pathGraph_val {n m : ℕ} (hnm : n ≤ m) (u : Fin n) :
+    (Hom.pathGraph hnm u).val = u.val := rfl
+
+
+/-- Create a walk from a path graph homomorphism. -/
+def Walk.ofPathGraphHom (G : SimpleGraph α) {n : ℕ} (hom : pathGraph (n + 1) →g G) :
+    G.Walk (hom ⊤) (hom ⊥) := by
+  induction n with
```

Comment:
This shouldn't be defined using the `induction` tactic. Also, I would rather the result be a `Walk` from the beginning to the end rather than the other way around.

To define this, you can make an auxiliary definition that creates a Walk from index `i` to the end, recursively (you probably need a `termination_by` to help it see that this is ok, or you could provide an additional length argument that decreases each iteration).